### PR TITLE
DataTracker + DataTrackerEntry

### PR DESCRIPTION
**Pull request by matthewperiut**
Modern Minecraft has a completely different interface for DataTracker and "DataTrackerEntry" as a class as I have it mapped is also not the same, so I would like some recommendations or suggestions if some names don't seem "yarny"

### CODE REVIEW

File: mappings/net/minecraft/entity/data/DataTracker.mapping [hunk 1]
```diff
@@ -1 +1,30 @@
 CLASS net/minecraft/class_216 net/minecraft/entity/data/DataTracker
+	FIELD field_1740 DATA_TYPE_MAP Ljava/util/HashMap;
+	FIELD field_1741 trackedData Ljava/util/Map;
```
Comment (mineLdiver) [inline mappings/net/minecraft/entity/data/DataTracker.mapping]: `entries` according to yarn - https://github.com/FabricMC/yarn/blob/4674b66184231baebe54ca8bf95568fd677f25f8/mappings/net/minecraft/entity/data/DataTracker.mapping#L3

File: mappings/net/minecraft/entity/data/DataTracker.mapping [hunk 2]
```diff
@@ -1 +1,30 @@
...
+		ARG 1 output
+	METHOD method_1507 getDirtyEntries ()Ljava/util/ArrayList;
+	METHOD method_1508 getInt (I)I
+		ARG 1 id
+	METHOD method_1509 set (ILjava/lang/Object;)V
+		ARG 1 id
+		ARG 2 object
+	METHOD method_1510 getString (I)Ljava/lang/String;
+		ARG 1 id
+	METHOD method_1511 updateEntries (Ljava/util/List;)V
```
Comment (mineLdiver) [inline mappings/net/minecraft/entity/data/DataTracker.mapping]: `writeUpdatedEntries` according to yarn - https://github.com/FabricMC/yarn/blob/4674b66184231baebe54ca8bf95568fd677f25f8/mappings/net/minecraft/entity/data/DataTracker.mapping#L14

File: mappings/net/minecraft/entity/data/DataTracker.mapping [hunk 3]
```diff
@@ -1 +1,30 @@
 CLASS net/minecraft/class_216 net/minecraft/entity/data/DataTracker
+	FIELD field_1740 DATA_TYPE_MAP Ljava/util/HashMap;
```
Comment (mineLdiver) [inline mappings/net/minecraft/entity/data/DataTracker.mapping]: Maybe `DATA_TYPES` is better

File: mappings/net/minecraft/entity/data/DataTrackerEntry.mapping [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+	FIELD field_1093 dataTypeId I
+	FIELD field_1094 id I
+	FIELD field_1095 value Ljava/lang/Object;
+	FIELD field_1096 dirty Z
+	METHOD <init> (IILjava/lang/Object;)V
+		ARG 1 dataTypeId
+		ARG 2 id
+		ARG 3 valuje
+	METHOD method_960 getId ()I
+	METHOD method_961 setValue (Ljava/lang/Object;)V
```
Comment (mineLdiver) [inline mappings/net/minecraft/entity/data/DataTrackerEntry.mapping]: `set` according to yarn - https://github.com/FabricMC/yarn/blob/4674b66184231baebe54ca8bf95568fd677f25f8/mappings/net/minecraft/entity/data/DataTracker.mapping#L46

File: mappings/net/minecraft/entity/data/DataTrackerEntry.mapping [hunk 5]
```diff
@@ -0,0 +1,17 @@
+CLASS net/minecraft/class_270 net/minecraft/entity/data/DataTrackerEntry
+	FIELD field_1093 dataTypeId I
+	FIELD field_1094 id I
+	FIELD field_1095 value Ljava/lang/Object;
+	FIELD field_1096 dirty Z
+	METHOD <init> (IILjava/lang/Object;)V
+		ARG 1 dataTypeId
+		ARG 2 id
+		ARG 3 valuje
```
Comment (mineLdiver) [inline mappings/net/minecraft/entity/data/DataTrackerEntry.mapping]: Spelling mistake

**Review by mineLdiver — CHANGES_REQUESTED** [review]

File: mappings/net/minecraft/entity/data/DataTrackerEntry.mapping (hunk 5 above)
Comment (matthewperiut) [reply mappings/net/minecraft/entity/data/DataTrackerEntry.mapping]: rip-

File: mappings/net/minecraft/entity/data/DataTracker.mapping (hunk 3 above)
Comment (matthewperiut) [reply mappings/net/minecraft/entity/data/DataTracker.mapping]: i agree

**Review by mineLdiver — APPROVED** [review]

**Review by DanyGames2014 — APPROVED** [review]
I don't have anything to say, looks good

**Review by calmilamsy — APPROVED** [review]
Looks good. I've looked at modern GUI code before, and this looks about right at a glance.

**Pull request merged**